### ISSUE TITLE
MWT have no base word set in the training data

### ISSUE THREAD

**Issue by AngledLuffa** [UniversalDependencies/UD_Marathi-UFAL#1]
There are some examples where the MWT expansions are blank in the training data, which is pretty bad for any models trying to learn from the dataset

In fact, it seems all of the MWT are split into _ with no actual text for the word pieces.

```
# sent_id = 3
# text = परंतु लाडामुळे तो बिघडला.
1       परंतु     परंतु     CCONJ   _       _       5       cc      _       _
2-3     लाडामुळे  _       _       _       _       _       _       _       _
2       _       लाड     NOUN    _       Case=Abs|Gender=Masc|Number=Sing        5       obl     _       _
3       _       मुळे      ADP     _       _       2       case    _       _
4       तो      तो      PRON    _       Case=Nom|Deixis=Remt|Gender=Masc|Number=Sing|Person=3   5       nsubj   _       _
5       बिघडला  बिघडणे   VERB    _       Aspect=Perf|Gender=Masc|Number=Sing|Person=3|VerbForm=Fin       0       root    _       SpaceAfter=No
6       .       .       PUNCT   _       _       5       punct   _       _
```

```
# sent_id = 10
# text = "आज त्याला हाकलून देणे तुला कठोरपणाचे वाटले तरी तेच हिताचे आहे," तो म्हणाला.
7-8     कठोरपणाचे        _       _       _       _       _       _       _       _
7       _       कठोरपणा NOUN    _       Case=Abs|Gender=Masc|Number=Sing        9       nmod:poss       _       _
8       _       चा      ADP     _       Gender=Neut|Number=Sing 7       case    _       _
9       वाटले    वाटणे    VERB    _       Aspect=Perf|Gender=Neut|Number=Sing|Person=3|VerbForm=Fin       0       root    _       _
10      तरी     तरी     SCONJ   _       _       13      mark    _       _
11-12   तेच      _       _       _       _       _       _       _       _
11      _       तो      PRON    _       Case=Nom|Deixis=Remt|Gender=Neut|Number=Sing|Person=3   13      nsubj   _       _
12      _       च       PART    _       _       11      discourse       _       _
13-14   हिताचे   _       _       _       _       _       _       _       _
13      _       हित     NOUN    _       Case=Abs|Gender=Neut|Number=Sing        9       ccomp   _       _
14      _       चा      ADP     _       Gender=Neut|Number=Sing 13      case    _       _
```

```
# sent_id = 12
# text = रात्रभर तिला झोप आली नाही.
1-2     रात्रभर  _       _       _       _       _       _       _       _
1       _       रात्र    NOUN    _       Case=Abs|Gender=Fem|Number=Sing 5       obl     _       _
2       _       भर      ADP     _       _       1       case    _       _
```

**Comment by dan-zeman** [comment]
Yes, I also noticed this problem a few days ago. It is already fixed in the dev branch (https://github.com/UniversalDependencies/UD_Marathi-UFAL/commit/c9c14ccf64eb69c5a250aa5f061afc08a1c6be9a).

**Comment by AngledLuffa** [comment]
Awesome, thanks!  I will apply it to our software right away.

Do you know this is the correct fix based on other Marathi researchers
sending that in?  I know absolutely nothing about Marathi.  I don't even
know if the text is always supposed to be whitespace separated or if people
knowhowtosegmentitsimilartoChinese.

**Comment by dan-zeman** [comment]
The fix was done by [a script that I wrote](https://github.com/udapi/udapi-python/blob/master/udapi/block/ud/mr/addformsinmwt.py), based on heuristics, so there might be room for improvement. On the other hand, for some types of multiword tokens there is no single correct solution.

Marathi normally uses spaces to delimit orthographic words, so this issue only applies to a limited number of orthographic words that should be understood as contractions of multiple syntactic words.

The most pervasive type of MWT in Marathi involves the possessive postposition चा|ची|चे|च्या (different forms of the postposition for different combinations of gender and number of the possessed entity), so it makes sense to cut off the postposition from the surface token, and leave whatever is left as the form of the noun. I suspect that it may not always be a form in which the noun would occur without the postposition, but the Marathi treebank reserves a case value for nominal forms in postpositional phrases, so it is probably OK to keep this form with this case value. However, there are also personal pronouns, which, when fused with the postposition, change their form more significantly. For example, 2SG (you) + POSS (= your) in Marathi means तू + चा|ची|चे|च्या _(tū + cā|cī|ce|cyā)_ but the result is not तूच्या _(tūcyā)_, it is actually तुझ्या _(tujhyā)_. Here I leave _tujh_ as the "to-be-used-with-postpositions" form of the pronoun.

**Comment by AngledLuffa** [comment]
Thanks for the explanation!  I am impressed by your knowledge of Marathi

**Comment by AngledLuffa** [comment]
Hmm, just discovered that the same thing is happening for 3 word MWTs and not yet labeled

There's a total of 9 in the train set, and I didn't find any in the dev or test set.

Example:

```
# sent_id = 34
# text = तिच्याभोवती तिची पाडसे खेळत होती.
# translit = ticyābhovatī ticī pāḍase kheḷata hotī.
1-3     तिच्याभोवती      _       _       _       _       _       _       _       Translit=ticyābhovatī
1       _       तो      PRON    _       Case=Abs|Deixis=Remt|Gender=Fem|Number=Sing|Person=3|PronType=Dem       7       obl     _       LTranslit=to
2       _       च्या     PART    _       _       1       case    _       LTranslit=cyā
3       _       भोवती   ADP     _       _       1       case    _       LTranslit=bhovatī
```

**Comment by dan-zeman** [comment]
Oh, I overlooked these, thanks for reporting! They slipped through because of different lemmatization of च्या. Fixed now in dev.
